airbyte-integrations/connectors/destination-s3-v2/src/test-integration/kotlin/io/airbyte/integrations/destination/s3_v2/S3V2WriteTest.kt

```diff
@@ -26,9 +40,38 @@ class S3V2WriteTest :
 }
 
 object S3V2DataDumper : DestinationDataDumper {
-    override fun dumpRecords(streamName: String, streamNamespace: String?): List<OutputRecord> {
-        // E2e destination doesn't actually write records, so we shouldn't even
-        // have tests that try to read back the records
-        throw NotImplementedError()
+    override fun dumpRecords(
+        spec: ConfigurationSpecification,
+        stream: DestinationStream
+    ): List<OutputRecord> {
+        val config =
+            S3V2ConfigurationFactory().makeWithoutExceptionHandling(spec as S3V2Specification)
+        val s3Client = S3ClientFactory.make(config)
+        // NOTE: This only works because I leave time out of the test config path
+        // We should probably have a test time provider that gives a fixed loaded at.
```

Comment:
that's not going to work for running in-docker :/

IMO this should just do the equivalent of `cat s3://bucket/namespace/name/**/*`, i.e. not even try to get the correct time? (I'm assuming the path format puts all the dynamically-generated stuff at the end)

Reply:
Fair point. I've got to add matching patterns to the variables soon when I get to managing destination state, at that point I'll change this to use those to fetch. For now this works. (I'll update the comment to reflect that.)

Reply:
* fixed the comment
* changed the interface for direct injection to take a nullable time provider
* added null guards to the variables to catch mistakenly relying on time-based variables at test time